src/main/java/com/example/demo/services/impl/UserServiceImpl.java

```diff
@@ -52,19 +68,34 @@ public User getUserById(final long id) {
 
     @Override
     public void addUser(final UserCreation user) {
+        if (user.getAge() < 0) {
+            log.error(ageValidError);
+            throw new RuntimeException(ageValidError);
+        }
         userRepository.save(userEntityMapper.map(user));
     }
 
     @Override
     public void updateUser(final UserUpdate userUpdate) {
-        final LocalDateTime creationDate = userRepository.getById(userUpdate.getId()).getCreatedAt();
-        final UserEntity updatedUserEntity = userEntityMapper.map(userUpdate);
-        updatedUserEntity.setCreatedAt(creationDate);
+        if (userUpdate.getAge() < 0) {
+            log.error(ageValidError);
```

Comment:
It is better to log user id as well to know exactly who was deleted/updated/added